File: src/main/webapp/src/app/model/import/import-progress.ts
import {ImportStatus} from './import-status.enum';

export type ImportProgress = {
  id: number;

  importedApplications: number;

  saveErrors: number;

  savedApplicants: number;

  totalCount: number;

  importStatus: ImportStatus;

  importedUids: number;

  notificationsSend: number;

  potentialDuplicates: number;

  error: string;
};

File: src/main/kotlin/pl/poznan/ue/matriculation/local/domain/import/ImportProgress.kt
package pl.poznan.ue.matriculation.local.domain.import

import pl.poznan.ue.matriculation.local.domain.enum.ImportStatus

interface ImportProgress {
    var id: Long?

    var importedApplications: Int

    var saveErrors: Int

    var savedApplicants: Int

    var totalCount: Int?

    var importedUids: Int

    var notificationsSend: Int

    var potentialDuplicates: Int

    var importStatus: ImportStatus

    var error: String?
}

